src/App/Fossa/VPS/Scan/RunWiggins.hs

```diff
@@ -0,0 +1,86 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.VPS.Scan.RunWiggins
+  ( execWiggins
+  , generateWigginsOpts
+  , WigginsOpts(..)
+  , ScanType(..)
+  )
+where
+
+import App.Fossa.VPS.Types
+import App.Fossa.VPS.EmbeddedBinary
+import Control.Carrier.Error.Either
+import Control.Effect.Diagnostics
+import Data.Functor (void)
+import Data.Text (Text)
+import qualified Data.Text as T
+import Effect.Exec
+import Path
+import Effect.Logger
+import Fossa.API.Types
+import App.Types
+import Text.URI
+
+data ScanType = ScanType
+  { scanSkipIpr :: Bool
+  , scanLicenseOnly :: Bool
+  }
+
+data WigginsOpts = WigginsOpts
+  { scanDir :: Path Abs Dir
+  , spectrometerArgs :: [Text]
+  }
+
+generateWigginsOpts :: Path Abs Dir -> Severity -> OverrideProject -> ScanType -> FilterExpressions -> ApiOpts -> ProjectMetadata -> WigginsOpts
+generateWigginsOpts scanDir logSeverity overrideProject scanType fileFilters apiOpts metadata =
+  WigginsOpts scanDir (generateSpectrometerArgs logSeverity overrideProject scanType fileFilters apiOpts metadata)
+
+generateSpectrometerArgs :: Severity -> OverrideProject -> ScanType -> FilterExpressions -> ApiOpts -> ProjectMetadata -> [Text]
+generateSpectrometerArgs logSeverity OverrideProject{..} ScanType{..} fileFilters ApiOpts{..} ProjectMetadata{..} =
+    "analyze"
+      : ["-endpoint", renderUri apiOptsUri, "-fossa-api-key", unApiKey apiOptsApiKey]
+      ++ optMaybeText "-name" overrideName
+      ++ optMaybeText "-revision" overrideRevision
+      ++ optMaybeText "-jira-project-key" projectJiraKey
+      ++ optMaybeText "-link" projectLink
+      ++ optMaybeText "-policy" projectPolicy
+      ++ optMaybeText "-project-url" projectUrl
+      ++ optMaybeText "-team" projectTeam
+      ++ optMaybeText "-title" projectTitle
+      ++ optBool "-license-only" scanLicenseOnly
+      ++ optBool "-skip-ipr-scan" scanSkipIpr
+      ++ optBool "-debug" (logSeverity == SevDebug)
+      ++ optFilterExpressions fileFilters
+      ++ ["."]
```

Comment:
[nit]: This seems a little simpler as a list literal.

```suggestion
  [ "analyze",
    "-endpoint", renderUri apiOptsUri,
    "-fossa-api-key", unApiKey apiOptsApiKey,
    optMaybeText "-name" overrideName,
    optMaybeText "-revision" overrideRevision,
    optMaybeText "-jira-project-key" projectJiraKey,
    optMaybeText "-link" projectLink,
    optMaybeText "-policy" projectPolicy,
    optMaybeText "-project-url" projectUrl,
    optMaybeText "-team" projectTeam,
    optMaybeText "-title" projectTitle,
    optBool "-license-only" scanLicenseOnly,
    optBool "-skip-ipr-scan" scanSkipIpr,
    optBool "-debug" (logSeverity == SevDebug),
    optFilterExpressions fileFilters,
    "."
  ]
```

Reply:
This doesn't seem to work given that we are evaluating type `[Text]` from these `opt*` expressions. Am I missing something?

Reply:
Ah, that's right.  I think there's a way to make this more readable, but it might take some thinking and I'd rather not block on it, it was a nit anyway.